src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcher.DefaultScheduler.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.Razor.Workspaces;
+
+internal abstract partial class ProjectSnapshotManagerDispatcher
+{
+    private sealed class DefaultScheduler : TaskScheduler, IDisposable
```

Comment:
`DefaultScheduler` is the real "meat" of this pull request. It creates an `AsyncQueue<Task>` and runs a method on the thread pool to process the queue.